common/logger/src/lib.rs

```diff
@@ -1,80 +1,40 @@
+//! This crates provides an API for logging in libra.
 //!
-//! # Text logging
+//! # Instrumenting with Logs
 //!
-//! Text logging is configured via `RUST_LOG` environment variable and has exactly same facade as the rust log crate.
+//! A set of logging macros (`info!`, `error!`, `warn!`, `debug!`, and `trace!`) is provided for
+//! emitting logs at different levels. All of these macros support the addition of providing
+//! structured data along with a formated text message.
 //!
-//! # Structured logging
-//!
-//! This crate contains two types of APIs for structured logging.
-//!
-//! 1) The `StructuredLogEntry` class and (`sl_info!`, `sl_error!`, ...) macros for directly composing structured logs.
-//! 2) A bridge between traditional `log!` macro and the structured logging API (which will be deprecated).
-//!
-//! ## Configuration
-//!
-//! Structured logger has separate log levels that are configured with `STRUCT_LOG_LEVEL`.
-//! It is set to `DEBUG` by default, but executes if structured logger is initialized.
-//!
-//! Structured logger can be initialized manually with one of the `init_XXX_struct_log()` functions.
-//! The preferred way to initialize structured logging is by using `init_struct_log_from_env()`.
-//! In this case, the `STRUCT_LOG_FILE` environment variable is used to set the file name for structured logs.
-//!
-//! ## Direct API
+//! ```
+//! use libra_logger::info;
 //!
-//! ```pseudo
-//! use std::collections::HashMap;
-//! use serde_json::Value;
+//! let world = "world!";
 //!
-//! pub struct StructuredLogEntry {
-//!     log: Option<String>,
-//!     pattern: Option<&'static str>,
-//!     category: Option<&'static str>,
-//!     name: Option<&'static str>,
-//!     module: Option<&'static str>,
-//!     location: Option<&'static str>,
-//!     timestamp: String,
-//!     level: Option<log::Level>,
-//!     data: HashMap<&'static str, Value>,
-//! }
-//!
-//! impl StructuredLogEntry {
-//!     pub fn new_named(category: &'static str, name: &'static str) -> Self { /* ... */ }
-//!     /* ... Builder style setters for chained initialization such as
-//!         entry.data(a, b).data(x, y) ... */
-//! }
+//! // formatted message
+//! info!("hello {}", world);
 //!
-//! // Usage:
-//! let logging_field: LoggingField<String> = LoggingField::new("String");
-//! let string = "test".to_string();
+//! // structured data can be logged using the format 'key = value'
+//! // where value implements Serialize
+//! let value1 = 5;
+//! info!(key1 = value1);
 //!
-//! info!(StructuredLogEntry::new_named("TransactionEvents", "Committed")
-//!    .data("block", &block)
-//!    .data_display("author", &author)
-//!    .field(&logging_field, &string)
-//!    .message(format!("Committed block: {:?} by {}", block, author))
+//! // You can even set multiple key/value pairs and a format message
+//! let value2 = false;
+//! info!(key1 = value1, key2 = value2, "hello {}", world);
```

Comment:
I would mention that anything after the format message will not be caught by the structured log key pairs.